mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -186,39 +186,49 @@ class LogisticRegression(override val uid: String)
     val states = optimizer.iterations(new CachedDiffFunction(costFun),
       initialWeightsWithIntercept.toBreeze.toDenseVector)
 
-    var state = states.next()
-    val lossHistory = mutable.ArrayBuilder.make[Double]
+    val (weights, intercept, lossHistory) = {
+      /*
+         Note that in Logistic Regression, the loss is log-likelihood which is invariance
```

Comment:
It would be nice to say "objective" to mean loss + regularization (and to stick with this in the future).